plugins/modules/sysrc.py

```diff
@@ -130,11 +131,12 @@ def exists(self):
         Use this dictionary to preform the tests.
         """
         (rc, out, err) = self.run_sysrc('-e', '-a')
-        conf = dict([i.split('=', 1) for i in out.splitlines()])
+        conf = dict((part.split('=', 1) for part in split(out, comments=True)))
+
         if self.value is None:
             return self.name in conf
         else:
-            return self.name in conf and conf[self.name] == '"%s"' % self.value
+            return self.name in conf and conf[self.name] == self.value
```

Comment:
This could be simplified to:
```suggestion
            return conf.get(self.name) == self.value
```